src/libraries/System.Text.Json/src/System/Text/Json/Serialization/DefaultReferenceResolver.cs

```diff
@@ -3,80 +3,62 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Diagnostics;
 
 namespace System.Text.Json.Serialization
 {
     /// <summary>
     /// The default ReferenceResolver implementation to handle duplicate object references.
     /// </summary>
-    /// <remarks>
-    /// It is currently a struct to save one unnecessary allcation while (de)serializing.
-    /// If we choose to expose the ReferenceResolver in a future, we may need to create an abstract class/interface and change this type to become a class that inherits from that abstract class/interface.
-    /// </remarks>
-    internal struct DefaultReferenceResolver
+    internal class DefaultReferenceResolver : ReferenceResolver
```

Comment:
If you name it DefaultPreserve**d**ReferenceResolver I think you should also use plural Reference**s**, which no longer matches the base type.

DefaultPreserveReferenceResolver (all singular) sounds a bit weird, I think it is fine as it is.

Reply:
FWIW PreserveReferenceResolver (preserve-reference resolver) seems okay to me. I don't think it needs to have "Default" in the name.

"DefaultReferenceResolver" is weird to me because this is being used with `ReferenceHandler.Preserve`, and this behavior is not the default.

Reply:
The way I see it is that this is the "Default" of `ReferenceHandler.Preserve`, it is not the default for `ReferenceHandler.Default`, such thing does not use a `ReferenceResolver` at all.

Reply:
Allright, I will go with your suggestion since it also matches with `ReferenceHandler.Preserve`.